aspnet/client-side/bundling-and-minification.rst

```diff
@@ -1,8 +1,301 @@
+
+Overview
+--------
+Because bundling combines multiple files into a single file, it reduces the number of requests to the server that is required to retrieve and display a web asset, such as a web page. You can create CSS, JavaScript and other bundles. Fewer files, means fewer HTTP requests from your browser to the server or from the service providing your application. This results in improved first page load performance.
+
+Minification performs a variety of different code optimizations to reduce the size of requested assets (such as CSS, image, JavaScript files). Common results of minification include removing unnecessary white space and comments, and shortening variable names to one character. 
+
+Consider the following JavaScript function:
+
+.. code-block:: javascript
+
+	AddAltToImg = function (imageTagAndImageID, imageContext) {
+		///<signature>
+		///<summary> Adds an alt tab to the image
+		// </summary>
+		//<param name="imgElement" type="String">The image selector.</param>
+		//<param name="ContextForImage" type="String">The image context.</param>
+		///</signature>
+		var imageElement = $(imageTagAndImageID, imageContext);
+		imageElement.attr('alt', imageElement.attr('id').replace(/ID/, ''));
+	}
+
+After minification, the function is reduced to the following:
+
+.. code-block:: javascript
+
+	AddAltToImg=function(t,a){var r=$(t,a);r.attr("alt",r.attr("id").replace(/ID/,""))};
+	
+In addition to removing the comments and unnecessary whitespace, the following parameters and variable names were renamed (shortened) as follows:
+
+==================  =======  
+Original            Renamed      
+==================  =======  
+imageTagAndImageID  t  
+imageContext        a  
+imageElement        r   
+==================  =======  
+
+Bundling and Minification in Visual Studio using Gulp
+-----------------------------------------------------
+You may be familiar with *runtime* bundling and minification using ASP.NET Web Optimization. In ASP.NET 5, you can bundle and minify your client-side resources during *design-time*. Using design-time bundling and minification, you build your minified files up front and then deploy them with your app as static files. By bundling and minifying up front, you have the advantage of fewer moving parts and reduced server load. However, it's important to recognize that design-time bundling and minification increases build complexity and only work with static files.
+
+Gulp is a JavaScript-based streaming build toolkit for client-side code. It is commonly used to stream client-side files through a series of processes when a specific event occurs in a build environment. The advantages of using gulp include automating common development processes, simplifying repetitive tasks, and speeding up overall development. For instance, you can use gulp to automate your minification processes or clean your development environment before a new build.
```

Comment:
use std def for gulp - first occurance should be a link to gulp - then next should be link to our gulp topic.
Gulp is a task/build runner for development. It allows you to do a lot of stuff within your development workflow. You can compile sass files, uglify and compress js files and much more. The kicker for gulp is that its a streaming build system which doesn't write temp files. It's like Pipes in bash.